packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -151,76 +152,103 @@ export class BlockSwitcher extends Component {
-											}
-										/>
-									</MenuGroup>
-								) }
 								{ possibleBlockTransformations.length !== 0 && (
-									<MenuGroup>
-										<div className="block-editor-block-switcher__label">
-											{ __( 'Transform to' ) }
-										</div>
-										<BlockTypesList
-											items={ possibleBlockTransformations.map(
-												( destinationBlockType ) => ( {
-													id:
-														destinationBlockType.name,
-													icon:
-														destinationBlockType.icon,
-													title:
-														destinationBlockType.title,
-												} )
+									<NavigableToolbar
+										orientation="vertical"
+										className="block-editor-block-switcher-toolbar"
+										role="menu"
+										/* translators: accessibility text for the block toolbar */
+										aria-label={ __( 'Block switcher' ) }
+									>
+										<MenuGroup>
+											<div className="block-editor-block-switcher__label">
+												{ __( 'Transform to' ) }
+											</div>
+											<BlockTypesList
+												items={ possibleBlockTransformations.map(
+													( {
+														name,
+														icon: destinationBlockTypeIcon,
+														title,
+													} ) => ( {
+														id: name,
+														icon: destinationBlockTypeIcon,
+														title,
+														role: 'menuitem', // this is to prevent scroll containers from scrolling (NavigableContainer internal check).
+													} )
+												) }
+												onSelect={ ( item ) => {
+													onTransform(
+														blocks,
+														item.id
+													);
+													onClose();
+												} }
+											/>
+										</MenuGroup>
+									</NavigableToolbar>
+								) }
+								{ hasBlockStyles && (
+									<NavigableToolbar
+										orientation="vertical"
+										className="block-editor-block-switcher-toolbar"
+										/* translators: accessibility text for the block toolbar */
+										aria-label={ __(
+											'Block Styles switcher'
+										) }
+									>
+										<MenuGroup>
+											<div className="block-editor-block-switcher__label">
+												{ __( 'Styles' ) }
+											</div>
+											<BlockStyles
+												clientId={
+													blocks[ 0 ].clientId
+												}
+												onSwitch={ onClose }
+												onHoverClassName={
+													this.onHoverClassName
+												}
+											/>
+											{ hoveredClassName !== null && (
+												<div className="block-editor-block-switcher__preview">
+													<div className="block-editor-block-switcher__preview-title">
+														{ __( 'Preview' ) }
+													</div>
+													<BlockPreview
+														viewportWidth={ 500 }
+														blocks={
+															hoveredBlockType.example
+																? getBlockFromExample(
+																		hoveredBlock.name,
+																		{
+																			attributes: {
+																				...hoveredBlockType
```

Comment:
This feel like very deep indentation. Is there a possibility to move it elsewhere?

Reply:
(Could be a variable defined earlier, or a separate component, for example.)